### PR TITLE
HINT: Refactor UI, cleanup let hint and add for support

### PR DESCRIPTION
**Pull request by farodin91**

### CODE REVIEW

**Comment by farodin91** [comment]
Any trick to add to test for loops without add stdlib

File: src/main/kotlin/org/rust/cargo/project/configurable/RustProjectConfigurable.kt [hunk 1]
```diff
@@ -156,15 +138,20 @@ class RustProjectConfigurable(
     override fun isModified(): Boolean {
         val settings = project.rustSettings
         val data = rustProjectSettings.data
+        var checkForChanges = false
+        for (option in hintProvider.supportedOptions) {
+            val checkbox = hintCheckboxes[option.id] ?: continue
+            if (checkbox.isSelected != option.get()) {
+                checkForChanges = true
```
Comment (matklad) [inline src/main/kotlin/org/rust/cargo/project/configurable/RustProjectConfigurable.kt]: We can  `return True` here instead using a mutable variable.

**Comment by farodin91** [comment]
@matklad Resolved

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**